service/application.go

```diff
@@ -254,6 +256,23 @@ func (app *Application) setupConfigurationComponents(ctx context.Context) error
 	}
 
 	app.service = service
+
+	// If config sources were used start a goroutine watching for updates.
+	if watchable, ok := app.parserProvider.(parserprovider.Watchable); ok {
+		go func() {
+			err := watchable.WatchForUpdate()
+			switch {
+			case errors.Is(err, configsource.ErrSessionClosed):
```

Comment:
If wanted we could add specific errors directly on parserprovider package so no import of configsource.

Reply:
Please add a TODO